sdk/typespec/Cargo.toml

```diff
@@ -24,3 +24,5 @@ thiserror.workspace = true
 default = ["http", "json"]
 http = ["dep:http-types"]
 json = ["dep:serde_json"]
+amqp_errors = []
+eventhubs_errors = []
```

Comment:
I know AMQP and Event Hubs aren't a perfect circle, but let's not get too granular. Can we just do "amqp"? Services that expose their own details - if they use the Azure error response "standard" (we need to improve the variations of this, granted) - customers can get that. Let's not pile on different services because I agree: that's a bad precedent.

Most (all?) other languages don't. .NET, for example, still just has the `RequestFailedException` for 99% of service-response errors.

Reply:
I also don't think it's outside the realm of reason for `azure_core_amqp` to have its own `Error` as well that can `into()` a `typespec::Error` as `Other` or maybe we need a per-service classification that uses a `kind(&self) -> 'static str` method.

That also means `azure_core_amqp` could define it's own shorthand `Result<T, azure_core_amqp::Error>` if it makes method signatures easier.

Reply:
Eventhubs is gone in current PR - Rick pointed out that typespec cannot have references to anything Azure specific, so Eventhubs goes away in favor of "Other".

I'm starting an email thread with you and Ashley about how we want to handle service specific (or protocol specific suites).

Reply:
Email or Teams? I'd prefer the latter: in our channel, in fact. Posterity and all.

No other language really has this. It comes down to the actionable comment below: what can devs actually do with this information? In most cases I suspect the answer is, at best: log/trace it. You don't need - or even want, I maintain - different kinds for that. Heck, look at Go: most of their errors are merely strings (perhaps with nested error messages, but still strings).

Error details for complexity's sake make for an unnecessarily complicated API.